packages/app-core/src/modules/commentsArea/usecases/searchCommentsAreas/searchCommentsAreas.ts

```diff
@@ -0,0 +1,38 @@
+import { createThunk } from '../../../../store/createThunk';
+import { selectIsFetchingCommentsAreas } from '../..';
+import { setCommentsAreasSearchQuery, setFetchingCommentsAreas } from '../../actions';
+import { fetchCommentsAreas } from '../index';
+
+const SEARCH_DEBOUNCE_TIMEOUT = 500;
+
+let searchTimeout: number | undefined;
+
+export const searchCommentsAreas = createThunk(async ({ dispatch, getState, timerGateway }, query: string) => {
+  if (searchTimeout) {
+    timerGateway.clearTimeout(searchTimeout);
+  }
+
+  if (!selectIsFetchingCommentsAreas(getState())) {
```

Comment:
Let's invert this condition and return if the comments areas list is already being fetched.

Reply:
Not possible due to the debounce being handled by the domain.